js/modules/k6/execution/execution_test.go

```diff
@@ -0,0 +1,79 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2021 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package execution
+
+import (
+	"context"
+	"testing"
+
+	"github.com/dop251/goja"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/js/common"
+	"go.k6.io/k6/js/modulestest"
+	"go.k6.io/k6/lib"
+	"go.k6.io/k6/stats"
+)
+
+func TestVUTags(t *testing.T) {
+	t.Parallel()
+
+	rt := goja.New()
+	ctx := common.WithRuntime(context.Background(), rt)
+	ctx = lib.WithState(ctx, &lib.State{
+		Options: lib.Options{
+			SystemTags: stats.NewSystemTagSet(stats.TagVU),
+		},
+		Tags: lib.NewTagMap(map[string]string{
+			"vu": "42",
+		}),
+	})
+	m, ok := New().NewModuleInstance(
+		&modulestest.InstanceCore{
+			Runtime: rt,
+			InitEnv: &common.InitEnvironment{},
+			Ctx:     ctx,
+		},
+	).(*ModuleInstance)
+	require.True(t, ok)
+	require.NoError(t, rt.Set("exec", m.GetExports().Default))
+
+	// overwrite a system tag is allowed
+	_, err := rt.RunString(`exec.vu.tags["vu"] = 101`)
+	require.NoError(t, err)
+	val, err := rt.RunString(`exec.vu.tags["vu"]`)
+	require.NoError(t, err)
+	assert.Equal(t, "101", val.String())
+
+	// not found
+	tag, err := rt.RunString(`exec.vu.tags["not-existing-tag"]`)
+	require.NoError(t, err)
+	assert.Equal(t, "undefined", tag.String())
+
+	// set
+	_, err = rt.RunString(`exec.vu.tags["custom-tag"] = "mytag"`)
+	require.NoError(t, err)
+
+	// get
+	tag, err = rt.RunString(`exec.vu.tags["custom-tag"]`)
+	require.NoError(t, err)
+	assert.Equal(t, "mytag", tag.String())
```

Comment:
Sorry, for being late to the party, but I missed this last week. 

I kind of feel like just going to string with w/e is given is the better approach. After all we are in javascript and IME this is what is happening everywhere in js land all the time. 

As you've noticed it's very likely that a user will like to set a number and having them get an exception(or silently skipping the set) is likely not the UX that is better. Arguably we can just support numbers and string, but I feel like just calling `String()` on the Value we get will be fine. We are doing the same (or just taking a `string` which under the hood does the same) in a lot of places and rarely had anything but the above problem of `[Object object]`. 

Also arguably this can be discussed in an issue and pulling the people who either write scripts on daily bases and/or people who write js to give their opinion ;)

Reply:
>Arguably we can just support numbers and string

I don't think we can do that, we are storing in a `map[string]string` so it would be impossible for the `Get` case to revert in the right type. Except, if we would move to `map[string]interface{}`.

@MStoykov @yorugac I checked the current tags system and apparently the following example is already supported:

```js
export default function () {
  http.get('http://httpbin.org', {tags: {my_tag: "I'm a tag"}});
  http.get('http://test.k6.io', {tags: {my_tag: 4}}); // implicit conversion here
}
```
so except if we would break the API or have an inconsistent behaviour across them, I think we should go with the original design having freedom and an implicit conversion to String.

Reply:
> I don't think we can do that, we are storing in a map[string]string so it would be impossible for the Get case to revert in the right type. Except, if we would move to map[string]interface{} but I don't think we need that complexity here.

I meant only from the perspective of setting it so that `exec.iteration.tags["something"] = 5` is fine, but looking at goja's documentation it will be kind of hard to make it and arguably, again, we already convert to string in a lot of places (including the relevant place for `tags` as you have noticed) so IMO we should just be calling `String()` (or `ToString()` not certain which one it was) and go with the flow of js :shrug:

Reply:
For my understanding, what about `val.ExportType()`? Can't that be used to get the `val is Object` condition? I've looked at `ExportType` only briefly but it seems it should return `map[string]interface{}` only for JS objects...
Btw, for arrays it should return `[]interface{}` apparently:
```golang
var (
	reflectTypeInt    = reflect.TypeOf(int64(0))
	reflectTypeBool   = reflect.TypeOf(false)
	reflectTypeNil    = reflect.TypeOf(nil)
	reflectTypeFloat  = reflect.TypeOf(float64(0))
	reflectTypeMap    = reflect.TypeOf(map[string]interface{}{})
	reflectTypeArray  = reflect.TypeOf([]interface{}{})
	reflectTypeString = reflect.TypeOf("")
)
```

Reply:
>Can't that be used to get the val is Object condition?

Ok, but in the hypothesis, we could get a response for this, what is the end goal? Why should we treat a JS Object in a different way from a Boolean or a Number?

Reply:
To help avoid garbage values in the tags. I.e. booleans and numbers will be implicitly converted to something sensible and recognizable, even if not always precise, but `[object Object]` as tag value is ~ 100% a mistake and not what user wants to see in the tags. 
I think if Objects are allowed here, we might end up having to add additional filtering on tags in other places, like output extensions, or in whatever implementation for tags appears during re-factoring.

Granted, the fact that this is already allowed in the normal tags is a point... :thinking: Though it leads to conclusion that it's one of the things to improve from UX perspective there too :slightly_smiling_face: But I guess that'd be out of scope of this PR so I can agree that this solution is fine here, for now at least.

Reply:
FWIW, we might want to do something similar to what @MStoykov did for adding values to custom metrics recently: https://github.com/grafana/k6/pull/1876#issuecomment-946753448 In that case, we expect only numbers. Adding anything that wasn't a number introduced problems, so now doing it will be discarded with a warning by default, but throw an exception if the `throw` option is enabled. 

In this case, we can probably be a bit more liberal and convert numbers and bools to string automatically. But @yorugac is completely right that having an object or an array here is ~100% a mistake we probably shouldn't silently accept. I am fine with making an issue about this and fixing it in a separate PR though. If we document that users are supposed to use only string/int/bool values, it won't be a breaking change.

Reply:
I kind of ... disagree here as there I was moving to numbers/booleans, but here we move to `string` and there is [Object.prototype.toString](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/toString), so it's technically possible to set a tag value that is an object or array or whatever and have the `toString` method of it be set to appropriate (either on the prototype of the thing or only on the particular instance) and that will be expected to lead to the method being called and used IMO ;).

But let's discuss this in an issue and again, maybe pulling some js people and people who write scripts to give their opinion

Reply:
:thinking:  true, you can define `toString` on your own objects, but as far as I know, there isn't a way to detect if the user has implemented it or if it will just default to the base `[object Object]`? So, overall, I think it's much safer to go with the explicit approach here, instead of silently converting everything to string. Users can easily call `.toString()` on their objects manually, if that's what they want, and the majority of the time trying to pass objects as strings is probably going to be a mistake. And, if someone complains, we can more easily relax the restriction than implement a new one... 

re: discussing this in an issue, what's the problem to discuss it here? This PR implement just a part of an issue, and this specific problem is is an implementation detail of this PR, so the PR is as good of a place to discuss it as any...

Reply:
@na-- `v0.34.1` is accepting something like:
```js
export default function () {
  http.get('http://httpbin.org', {tags: {my_tag: {name: "bob", age: 25}}});
}

// http_req_duration.my_tag
// [object Object]
```
Doing Discard+Warning in this PR will generate something not consistent with the previous case. Should we change both then?

Reply:
:disappointed: :thinking: 

For this PR, I  think you should just make `exec.vu.tags` work like this, everything else is out of scope. This is a new feature and it won't be a breaking change to have more strict validation when it's launched.

For the `tags` parameter in HTTP requests, custom metrics, etc., please make a new issue to discuss how, if and when we should address it. We might decide to follow @MStoykov's suggestion and just do nothing, continue silently casting tag values to strings. Then we can also relax the validation here, if we decide to. Or we might decide to make the minor breaking change and introduce the stricter validation there as well :man_shrugging: But it definitely deserves more discussion and input from JS developers, and is definitely out of scope for this PR.

Reply:
ok, for summary:
* we accept `string`, `bool`, `number`
* if another kind is provided then discard and warn about it
* if `--throw` then raise an exception

Reply:
:+1:, + create a new issue to discuss this further, potentially adressing the same problem in other places or relaxing this validation here

Reply:
New issue is https://github.com/grafana/k6/issues/2218, resolving this thread